libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java

```diff
@@ -135,6 +139,33 @@ public static String getSHA256Hash(String privateKey) {
         return hashedString;
     }
 
+    /**
+     * Generate a RSA-2048 keypair and get the encoded public key string
+     *
+     * @return RSA-2048 public key string
+     */
+    @Nullable
+    public static synchronized String getRSA2048PublicString(String name) {
+        final SharedPreferences prefs = SalesforceSDKManager.getInstance().getAppContext().getSharedPreferences(SHARED_PREF_FILE, 0);
+        final String id = prefs.getString(getSharedPrefKey(name), null);
+
+        if (id != null) {
+            UNIQUE_IDS.put(name, id);
```

Comment:
In line with the above comment, let's use a new `HashMap`, not reuse `UNIQUE_IDS`. Remember there could be collisions in `name` and the algorithm used is different.